code/game/gamemodes/vampire/vampire_powers.dm

```diff
@@ -0,0 +1,347 @@
+/obj/effect/proc_holder/spell
+	var/gain_desc
+	var/blood_used = 0
+	var/vamp_req = FALSE
+
+/datum/vampire_passive
+	var/gain_desc
+
+/datum/vampire_passive/New()
+	..()
+	if(!gain_desc)
+		gain_desc = "You have gained \the [src] ability."
+
+
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+/datum/vampire_passive/regen
+	gain_desc = "Your rejuvination abilities have improved and will now heal you over time when used."
+
+/datum/vampire_passive/vision
+	gain_desc = "Your vampiric vision has improved."
+
+/datum/vampire_passive/full
+	gain_desc = "You have reached your full potential and are no longer weak to the effects of anything holy and your vision has been improved greatly."
+
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+/obj/effect/proc_holder/spell/self/rejuvenate
+	name = "Rejuvenate"
+	desc= "Flush your system with spare blood to remove any incapacitating effects."
+	action_icon_state = "rejuv"
+	charge_max = 200
+	stat_allowed = 1
+	action_background_icon_state = "bg_demon"
+	vamp_req = TRUE
+
+/obj/effect/proc_holder/spell/self/rejuvenate/cast(list/targets, mob/user = usr)
+	var/mob/living/carbon/U = user
+	U.stuttering = 0
+
+	var/datum/antagonist/vampire/V = U.mind.has_antag_datum(ANTAG_DATUM_VAMPIRE)
+	if(!V) //sanity check
+		return
+	for(var/i = 1 to 5)
+		U.adjustStaminaLoss(-10)
+		if(V.get_ability(/datum/vampire_passive/regen))
+			U.adjustBruteLoss(-1)
+			U.adjustOxyLoss(-2.5)
+			U.adjustToxLoss(-1)
+			U.adjustFireLoss(-1)
+		sleep(7.5)
+
+
+/obj/effect/proc_holder/spell/targeted/hypnotise
```

Comment:
i'd ask to make it cursor-based, but that has less priority

there's the issue that this spell is what made people hate shadowlings